postgres/configure-and-start-postgres.sh

```diff
@@ -0,0 +1,142 @@
+    echo "            --password <USER_PWD> --data-dir <PATH>"
+    echo
+    echo "Options:"
+    echo "   --host <HOST>          Host IP to accept connections on"
+    echo "   --dbname <DBNAME>      Database name to create"
+    echo "   --username <USERNAME>  Role name to create"
+    echo "   --password <USER_PWD>  Password to use for new user"
+    echo "   --data-dir <PATH>      Postgres data dir path (default: Postgres default)."
+    exit 1
+}
+
+HOST=
+DBNAME=
+USERNAME=
+PASSWORD=
+DATA_DIR=
+
+# Let's get the command line arguments.
+while [ $# -gt 0 ]; do
+    if [ "${1}" = "--host" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        HOST="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--dbname" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        DBNAME="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--username" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        USERNAME="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--password" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        PASSWORD="${1}"
+        shift; continue
+    fi
+    if [ "${1}" = "--data-dir" ]; then
+        shift; if [ -z "${1}" ]; then usage; fi
+        DATA_DIR="${1}"
+        shift; continue
+    fi
+    echo -e "\nParameter '$1' is not recognized\n"
+    usage
+done
+
+if [ -z "${HOST}" ] || [ -z "${DBNAME}" ] || [ -z "${USERNAME}" ] || [ -z "${PASSWORD}" ]; then
+    usage
+fi
+
+
+#####################################
+### Actual installation procedure ###
+#####################################
+
+${SCRIPT_DIR}/install.sh
+
+log "Updating Postgres configs to accept host connections to the Database"
+PG_MAJOR_VER=$(psql --version | awk '{ print $3 }' | awk -F '.' '{ print $1 }')
+PG_VERSION=$(psql --version | awk '{ print $3 }' | awk -F '.' '{ print $1 "." $2 }')
+
+if [ "${PG_MAJOR_VER}" = "10" ]; then
+    PG_CONFIG_DIR="/etc/postgresql/${PG_MAJOR_VER}"
+else
+    PG_CONFIG_DIR="/etc/postgresql/${PG_VERSION}"
+fi
+
+PG_CONF="${PG_CONFIG_DIR}/main/postgresql.conf"
+PG_HBA="${PG_CONFIG_DIR}/main/pg_hba.conf"
+
+# Configure postgres to listen on the specified host
+if grep -q -E "^listen_addresses *=" "${PG_CONF}"
+then
+    sed -i "s/^listen_addresses *=.*/listen_addresses = 'localhost,${HOST}'/" "${PG_CONF}"
+else
+    echo "listen_addresses = 'localhost,${HOST}'" >> "${PG_CONF}"
+fi
+
+# Configure postgres to use specific data directory
+if [ -n "${DATA_DIR}" ]; then
+    if grep -q -E "^data_directory *=" "${PG_CONF}"
+    then
+        sed -i "s/^data_directory *=.*/data_directory = '${DATA_DIR}'/" "${PG_CONF}"
+    else
+        echo "data_directory = '${DATA_DIR}'" >> "${PG_CONF}"
+    fi
+fi
+
+# Allow host connections to the specified DB
+if grep -q -E "^host[ \t]+${DBNAME}[ \t]+${USERNAME}[ \t]+" "${PG_HBA}"
+then
+    sed -i "s|^host[ \t]+${DBNAME}[ \t]+${USERNAME}[ \t]+.*|host ${DBNAME} ${USERNAME} ${HOST}/0 md5|" "${PG_HBA}"
```

Comment:
Should this be `0.0.0.0/0` rather than `${HOST}/0`?

Reply:
You're right. I copied this mistake from TaskQueue e2e test. Probably initial intention was to limit access to the server from outside private network, but then we'd need to use mask like 24.
I'll change it to `0.0.0.0/0` for now as mask of 24 bits won't be working for all networks.